---
title: "HEK293T_Mengrui"
author: "MVI"
date: "2024-02-02"
output: html_document
---

Load in packages
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("DESeq2")
BiocManager::install("arrayQualityMetrics")
install.packages("gplots")
install.packages("pheatmap")
install.packages("vegan")
install.packages("ggplot2")
install.packages("ggrepel")
install.packages("VennDiagram")
install.packages("ggtext")
install.packages("ggpubr")
install.packages("reshape2")
install.packages("tidyselect")
install.packages("dplyr")

library(DESeq2)
library(arrayQualityMetrics)
library(gplots)
library(pheatmap)
library(vegan)
library(ggplot2)
library(ggrepel)
library(VennDiagram)
library(ggtext)
library(ggpubr)
library(reshape2)
library(tidyselect)
library(dplyr)
```

Set your working directory (the entire path to the folder that contains this rmarkdown and your data files)
```{r}
knitr::opts_knit$set(root.dir = "/Users/mariaingersoll/Desktop/BU_Research/Davies-Gilmore/Mengrui_HEK293T/HEK293T_RNAseq")
#using the knitr option, it ensures that all chunks run in this rmd are performed in this working directory

#check your working directory:
getwd()
```

Open and inspect counts data (preprocessed using "tagseq_processing_HOWTO" on BU SCC)
```{r}
#load the count file into R and name it "countData"
countData = read.table("./Mengrui_HEK293T_2024_counts.txt")

#how long is the file (in other words, how many genes do you have count data for)?
length(countData[,1])
#59048

#look at the first few rows of the your file
head(countData)

#rename each of your samples (each column) with shorter names
newColNames = c("flag1_S33", "flag2_S34", "flag3_S35", "N1_S36", "N2_S37", "N3_S38", "NLS1_S39", "NLS2_S40", "NLS3_S41")
colnames(countData)=paste(newColNames)
length(countData[,1])
#59048
head(countData)

#look at the distribution of your count data throughout your samples
totalCounts=colSums(countData)
barplot(totalCounts, col=c("mediumblue", "mediumblue", "mediumblue", "red", "red", "red", "darkgreen", "darkgreen", "darkgreen"), ylab="Raw Counts", xlab="HEK293T")


min(totalCounts)
#480525     
max(totalCounts)
#862648 
```

Need to make a metadata file with information about each sample and its respective condition
```{r}
sample=c("flag1_S33", "flag2_S34", "flag3_S35", "N1_S36", "N2_S37", "N3_S38", "NLS1_S39", "NLS2_S40", "NLS3_S41")
s=data.frame(sample)
sampledata <-s
sampledata

treat=c("Flag", "Flag", "Flag", "N", "N", "N", "NLS", "NLS", "NLS")
g=data.frame(treat)
colData<- g
colData

allData = cbind(sampledata, colData)
allData
```

Run preliminary DESeq (differential expression testing) and outlier analysis (using arrayqualitymetrics)
```{r}
dds<-DESeqDataSetFromMatrix(countData=countData, colData=colData, design=~treat)
dds<-DESeq(dds)

vsd.ge=assay(vst(dds))
rl=vst(dds)
e=ExpressionSet(assay(rl), AnnotatedDataFrame(as.data.frame(colData(rl))))
arrayQualityMetrics(e,outdir="/Users/mariaingersoll/Desktop/BU_Research/Davies-Gilmore/Mengrui_HEK293T/HEK293T_RNAseq/arrayqualitymetrics",intgroup=c("treat"),force=T)
```
Open the index.html file that is printed into the output directory
--> no outliers

Visualize the results
`results` extracts the results table from the DESeq analysis and the DESeqDataSet is used to generate the dispersion plot.
  --> Have to compare each of your "treatments" to the control
  --> And then the "treatments" to each other too
  
```{r}
#N to Flag
head(dds)
res_N_Flag<- results(dds, contrast = c("treat", "N", "Flag"))
res_N_Flag

#NLS to Flag
res_NLS_Flag<- results(dds, contrast = c("treat", "NLS", "Flag"))
res_NLS_Flag

#N to NLS
res_N_NLS<- results(dds, contrast = c("treat", "N", "NLS"))
res_N_NLS
```

The above code gives the log2 fold change (MLE) of a treatment against a baseline
For example, in res_N_Flag, a pos log2FC values indicates that a gene is upregulated in N compared to Flag
-->The name provided in the second element of the results() command is the baseline


Dispersion plot
Read more about dispersion here: https://hbctraining.github.io/DGE_workshop/lessons/04_DGE_DESeq2_analysis.html

Basically, as the mean of normalized counts for a given gene increases, the dispersion (variance) should decrease
```{r}
plotDispEsts(dds, main="Dispersion Uptake")
```

- Now we are going to retrieve the rlog data, which will give us better transformation when size factors vary across samples
- regularized log transformation, transforms COUNT DATA to log2 scale to minimize differences between samples for rows with small counts
```{r}
rld <- rlogTransformation(dds, blind=TRUE)
head(assay(rld))
hist(assay(rld))
```

- The assay function allows you to access the matrix-like data, so that you can view it bc head(rld) gives you just a summary
- The histogram above displays the frequency of each binned rld count value

Raw Sample Comparison

- Making a sample distance heatmap
- as.matrix attempts to turn its argument into a matrix
- dist computes and returns the distance matrix computed using the specified distance measure to compute distances between the rows of a data matrix (assay(rld))
- Create a sample heatmap of the relatedness based on the count distance/difference between samples
```{r}
sampleDists <- as.matrix(dist(t(assay(rld))))

heatmap.2(as.matrix(sampleDists), key=F, trace="none",
          col=colorpanel(100, "black", "white"), ColSideColors=c("mediumblue",  "mediumblue", "mediumblue", "red", "red", "red", "darkgreen", "darkgreen", "darkgreen"),RowSideColors = c("mediumblue", "mediumblue", "mediumblue", "red", "red", "red", "darkgreen", "darkgreen", "darkgreen"),
          margin=c(10, 10))

#saved as sampledisthm.pdf as landscape with 5x5 dimensions

#all the flags are grouping together and then the Ns and the NLS are grouping together
```

Look at number of differentially expressed genes between the different treatments
Use a p-adjusted value of <0.05
```{r}
head(res_N_Flag)
head(res_NLS_Flag)
head(res_N_NLS)

table(res_N_Flag$padj<0.05)
# FALSE=9587 TRUE=1379
summary(res_N_Flag, alpha=0.05)
# out of 59048 with nonzero total read count
#adjusted p-value < 0.05
#LFC > 0 (up)       : 626, 1.1%
#LFC < 0 (down)     : 753, 1.3%
#outliers [1]       : 0, 0%
#low counts [2]     : 48082, 81%
#(mean count < 5)

table(res_NLS_Flag$padj<0.05)
# FALSE=7994 TRUE=1828 
summary(res_NLS_Flag, alpha=0.05)
#out of 59048 with nonzero total read count
#adjusted p-value < 0.05
#LFC > 0 (up)       : 743, 1.3%
#LFC < 0 (down)     : 1085, 1.8%
#outliers [1]       : 0, 0%
#low counts [2]     : 49226, 83%
#(mean count < 5)

table(res_N_NLS$padj<0.05)
# FALSE=59047 TRUE=1
summary(res_N_NLS, alpha=0.05)
#out of 59048 with nonzero total read count
#adjusted p-value < 0.05
#LFC > 0 (up)       : 1, 0.0017%
#LFC < 0 (down)     : 0, 0%
#outliers [1]       : 0, 0%
#low counts [2]     : 0, 0%
#(mean count < 0)

plotMA(res_N_Flag, main="N vs Flag")
#saved as plotma_N_Flag as landscape 5x4 in Figures
plotMA(res_NLS_Flag, main="NLS vs Flag")
#saved as plotma_NLS_Flag as landscape 5x4 in Figures
plotMA(res_N_NLS, main="N vs NLS")
#saved as plotma_N_NLS as landscape 5x4 in Figures
```

Now make a table of the differential expression values (log2fc) between N, Flag; NLS, Flag; and N, NLS
```{r}
write.table(res_N_Flag, file="./Data_Files/N_Flag_DE_24.txt", quote=F, sep="\t")
head(read.delim("./Data_Files/N_Flag_DE_24.txt"))
res_N_Flag<- read.delim("./Data_Files/N_Flag_DE_24.txt")
head(res_N_Flag)
res_N_Flag$lfc_NFlag <- res_N_Flag$log2FoldChange
lfc_N_Flag <- res_N_Flag %>%
  dplyr::select(lfc_NFlag)
head(lfc_N_Flag)

write.table(res_NLS_Flag, file="./Data_Files/NLS_Flag_DE_24.txt", quote=F, sep="\t")
head(read.delim("./Data_Files/NLS_Flag_DE_24.txt"))
res_NLS_Flag<- read.delim("./Data_Files/NLS_Flag_DE_24.txt")
head(res_NLS_Flag)
res_NLS_Flag$lfc_NLSFlag <- res_NLS_Flag$log2FoldChange
lfc_NLS_Flag <- res_NLS_Flag %>%
  dplyr::select(lfc_NLSFlag)
head(lfc_NLS_Flag)

write.table(res_N_NLS, file="./Data_Files/N_NLS_DE_24.txt", quote=F, sep="\t")
head(read.delim("./Data_Files/N_NLS_DE_24.txt"))
res_N_NLS<- read.delim("./Data_Files/N_NLS_DE_24.txt")
head(res_N_NLS)
res_N_NLS$lfc_NNLS <- res_N_NLS$log2FoldChange
lfc_N_NLS <- res_N_NLS %>%
  dplyr::select(lfc_NNLS)
head(lfc_N_NLS)
```

-Make a table with logfc in NvsFlag and lfc in NLSvsFlag as columns, correlate two columns, is there a pos linear relationshop in lfc? 
```{r}
lfc <- cbind(lfc_N_Flag, lfc_NLS_Flag)
head(lfc)

ggscatter(lfc, x = "lfc_NFlag", y = "lfc_NLSFlag", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "DEG LogFoldChange N vs Flag", ylab = "DEG LogFoldChange NLS vs Flag")

#saved as correlation_NNLS.pdf as landscape 5x4 in Figures
#positively correlated R=0.64, p<2.2e-16
```

- Get the p-values from your data file
- Create val_N_Flag that is a table from res_N_Flag of just the pvalues and adjusted p-values (p-adj)
- Then give the columns appropriate names
```{r}
val_N_Flag=cbind(res_N_Flag$pvalue, res_N_Flag$padj)
head(val_N_Flag)
colnames(val_N_Flag)=c("pval.N_Flag", "padj.N_Flag")
length(val_N_Flag[,1])
# 59048

val_NLS_Flag=cbind(res_NLS_Flag$pvalue, res_NLS_Flag$padj)
head(val_NLS_Flag)
colnames(val_NLS_Flag)=c("pval.NLS_Flag", "padj.NLS_Flag")
length(val_NLS_Flag[,1])
# 59048

val_N_NLS=cbind(res_N_NLS$pvalue, res_N_NLS$padj)
head(val_N_NLS)
colnames(val_N_NLS)=c("pval.N_NLS", "padj.N_NLS")
length(val_N_NLS[,1])
# 59048
```

Return a logical vector in which both cases (pval and padj) are complete (no missing values)
```{r}
table(complete.cases(val_N_Flag))
# FALSE=48082 TRUE=10966

table(complete.cases(val_NLS_Flag))
# FALSE=49226 TRUE=9822

table(complete.cases(val_N_NLS))
# TRUE=59048
```

- Make a table (that you can visualize via assay) of the rlogdata and pvals
- Then make the column names for rld the same ones that were in colData$treat
- And separate out into N, NLS, and Flag
```{r}
rlog=rlogTransformation(dds, blind=TRUE) 
rld=assay(rlog)
head(rld)
length(rld[,1])
#59048

rld_Flag <- rld[,1:3]
head(rld_Flag)

rld_N <- rld[,4:6]
head(rld_N)

rld_NLS <- rld[,7:9]
head(rld_NLS)
```

- Combine rlds with your tables of pval and padj values (with the respective conditions)
```{r}
rldpvals_NFlag=cbind(rld_N,val_N_Flag)
head(rldpvals_NFlag)
dim(rldpvals_NFlag)
table(complete.cases(rldpvals_NFlag))
#FALSE  TRUE 
#48082 10966
write.csv(rldpvals_NFlag, "./Data_Files/mingers_NFlag_RLDandPVALS_24.csv", quote=F)

rldpvals_NLSFlag=cbind(rld_NLS,val_NLS_Flag)
head(rldpvals_NLSFlag)
dim(rldpvals_NLSFlag)
table(complete.cases(rldpvals_NLSFlag))
#FALSE  TRUE 
#49226  9822 
write.csv(rldpvals_NLSFlag, "./Data_Files/mingers_NLSFlag_RLDandPVALS_24.csv", quote=F)
```

Read back in the csv files you just made to prep for PCA
```{r}
rldpvals_NFlag <- read.csv(file="./Data_Files/mingers_NFlag_RLDandPVALS_24.csv", row.names=1)
head(rldpvals_NFlag)
rld_NFlag <- rldpvals_NFlag[,1:3]
head(rld_NFlag)

rldpvals_NLSFlag <- read.csv(file="./Data_Files/mingers_NLSFlag_RLDandPVALS_24.csv", row.names=1)
head(rldpvals_NLSFlag)
rld_NLSFlag <- rldpvals_NLSFlag[,1:3]
head(rld_NLSFlag)
```

Now going to perform PCA on the data to visualize overall effect of experimental covariates and batch effects
- t transposes the rld table and makes all the rows columns, and the columns rows. So here the isoform names that were rownames are now the column names. 
- prcomp performs a pca on given data matrix and returns results as an object of class prcomp
- sdev is the standard deviations of the principal components
- Using the sdev, calculate the proportion that each PC corresponds to the variance
- Then round PC1 and PC2 (times 100, to 1 sigfig)
- x from prcomp seems like it's the coordinates of each treatment in the PCA
- turn pca$x into a dataframe
```{r}
#for ALL
rld_t=t(rld)
#colnames(rld_t)
pca <- prcomp(rld_t,center = TRUE)
li <- pca$sdev^2 / sum(pca$sdev^2)
pc1v <- round(li[1] * 100, 1)
pc2v <- round(li[2] * 100, 1)
pca_s <- as.data.frame(pca$x)
pca_s <- pca_s[,c(1,2)]

pca_s$Samples = row.names(pca_s)
pca_s$sample = c("flag1_S33", "flag2_S34", "flag3_S35", "N1_S36", "N2_S37", "N3_S38", "NLS1_S39", "NLS2_S40", "NLS3_S41")
pca_s$treat=allData$treat

head(pca_s)
```

Creating PCA plots
```{r}
treat_color= c("mediumblue", "red", "darkgreen")
#fill_color=c("white","white","gray50","purple")
#sym_shape = c(21,22,23,24,25,21,22,23,24,25)


pca_plot <- ggplot(pca_s, aes(PC1, PC2, color = treat, group=treat)) +
  geom_point(size=2) +
  #  geom_text_repel(aes(label=Samples)) +
  scale_colour_manual(values=treat_color)+
  #scale_fill_manual(values=fill_color, guide=NULL)+
  #scale_shape_manual(values=sym_shape, guide=guide_legend(override.aes = list(fill=list("white","white","white","white","white","black","black","black","black","black")))) +
  stat_ellipse()+
  labs(color="Treatment")+
  xlab(paste0("PC1: ",pc1v,"% variance")) +
  ylab(paste0("PC2: ",pc2v,"% variance")) +
  theme(panel.background = element_rect(color="black", fill="white"),
        axis.text = element_text(colour="black", size=10), 
        axis.title.x = ggtext::element_markdown(face="bold", size=12),
        axis.title.y = ggtext::element_markdown(face="bold",size=12),
        panel.grid.major = element_line(colour = "gray80", linewidth = 0.5, linetype="dotted"),
        legend.position="right",
        legend.background=element_rect(color="black"),
        legend.text=element_text(face="bold", color="black", size=8),
        legend.title= element_blank(),
        legend.text.align = 0,
        legend.key.size=unit(10,"point"))
pca_plot # saved as portrait with 4x3 as pca_012223.pdf in Figures

```


