---
title: "HEK293T_GO_GSEA"
output: html_document
date: "2024-03-14"
editor_options: 
  chunk_output_type: inline
---
THE GSEA CODE ONLY WORKS WITH R 4.3


```{r}
setwd("/projectnb/davies-hb/Maria/Mengrui_HEK293T/HEK293T_RNAseq")
```

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("DESeq2")
BiocManager::install("arrayQualityMetrics")
install.packages("gplots")
install.packages("pheatmap")
install.packages("vegan")
install.packages("ggplot2")
install.packages("ggrepel")
install.packages("VennDiagram")
install.packages("ggtext")
install.packages("reshape2")
install.packages("tidyselect")
install.packages("dplyr")
install.packages("devtools")
install.packages("tibble")
install.packages("RColorBrewer")
```


```{r}
library(DESeq2)
library(arrayQualityMetrics)
library(gplots)
library(pheatmap)
library(vegan)
library(ggplot2)
library(ggrepel)
library(VennDiagram)
library(ggtext)
library(reshape2)
library(tidyselect)
library(dplyr)
library(devtools)
library(tibble)
library(RColorBrewer)
```

read in iso2gene
NAH
```{r}
iso2gene = read.table("./Data_Files/GRCh38_gene2iso.tab", sep="\t")
head(iso2gene)

colnames(iso2gene)=paste(c("DESCRIPTION","NAME"))
head(iso2gene)
```

Read in gtf to make iso2gene
```{r}
#read in gtf to try to get a better iso2gene
gtf <- read.table("/projectnb/davies-hb/Maria/Mengrui_HEK293T/GRCh38_latest_genomic.gtf", sep="\t")
head(gtf)
iso2gene_good <- gtf[,c(1,9)]
head(iso2gene_good)
iso2gene_good = iso2gene_good %>%
  mutate(NAME = gsub(".*gene_name ", "", V9)) %>%
  mutate(NAME = gsub(";", "", NAME)) %>%
  mutate(ISO = gsub(".*transcript_id rna-", "", V9)) %>%
  mutate(ISO = gsub("; .*", "", ISO)) %>%
  #mutate(ISO = gsub("transcript_.*", "", ISO))%>%
  dplyr::select(NAME, ISO) %>%
  distinct()
#head(seq2iso2gene)
#nrow(seq2iso2gene)
#seq2iso <- seq2iso2gene %>%
#  dplyr::select(V1, ISO)
#length(unique(seq2iso$ISO))

head(iso2gene_good)

seq2iso <-  read.table("./Data_Files/GRCh38_seq2iso_new.tab", sep=" ")
head(seq2iso)
nrow(seq2iso)
#185121

iso2gene_sep <- subset(iso2gene_good, ISO %in% seq2iso$V1)
nrow(iso2gene_sep)
head(iso2gene_sep)

gff <- read.table("/projectnb/davies-hb/Maria/Mengrui_HEK293T/GRCh38_latest_genomic.gff", sep="\t")
head(gff)
length(unique(gff$V1))
#512
gff_cds <- subset(gff, V3=="CDS")
gff_cds
nrow(gff_cds)
#1520592

cds_seq2iso_1 <- gff_cds %>%
  mutate(REFSEQ = gsub(".*ID=cds-", "", V9))%>%
  mutate(REFSEQ = gsub(";.*", "", REFSEQ))%>%
  mutate(ISO = gsub(".*Parent=rna-", "", V9)) %>%
  mutate(ISO = gsub(";.*", "", ISO)) %>%
  dplyr::select(REFSEQ, ISO)%>% 
  relocate(ISO, .before=REFSEQ) %>%
  distinct()

cds_seq2iso <- gff_cds %>%
  mutate(REFSEQ = gsub(".*ID=cds-", "", V9))%>%
  mutate(REFSEQ = gsub("\\..*", "", REFSEQ))%>%
  mutate(ISO = gsub(".*Parent=rna-", "", V9)) %>%
  mutate(ISO = gsub(";.*", "", ISO)) %>%
  dplyr::select(REFSEQ, ISO)%>% 
  distinct()

cds_seq2seq <- gff_cds %>%
  mutate(REFSEQ = gsub(".*ID=cds-", "", V9))%>%
  mutate(REFSEQ = gsub(";.*", "", REFSEQ))%>%
  mutate(REFSEQ_new = gsub(".*ID=cds-", "", V9))%>%
  mutate(REFSEQ_new = gsub("\\..*", "", REFSEQ_new))%>%
  dplyr::select(REFSEQ, REFSEQ_new)%>% 
  distinct()

length(unique(cds_seq2iso$REFSEQ)) 
#115178
length(unique(cds_seq2iso$ISO))
#119520

write.table(cds_seq2iso_1, file="./Data_Files/cds_seq2iso_input.tab", quote=F, sep="\t", row.names = FALSE, col.names = FALSE)

#shared_both_col <- as.data.frame(shared_both_col) %>% 
#  left_join(cds_seq2seq)
#shared_both_col <- shared_both_col %>%
#  left_join(cds_seq2iso_1)
#write.csv(shared_both_col, file = "./Data_Files/shared_both_col.csv", quote=F, row.names=FALSE)
```


```{r}
#########
res_N_Flag_new<- read.delim("./Data_Files/N_Flag_DE_24_new.txt")
head(res_N_Flag_new)
res_N_Flag_new$lfc_NFlag <- res_N_Flag_new$log2FoldChange
head(res_N_Flag_new)

res_NLS_Flag_new<- read.delim("./Data_Files/NLS_Flag_DE_24_new.txt")
head(res_NLS_Flag_new)
res_NLS_Flag_new$lfc_NLSFlag <- res_NLS_Flag_new$log2FoldChange
head(res_NLS_Flag_new)
```



Get each of your DEG lists (padj<0.05 no lfc filter, padj<0.05 and lfc>|1.5|, padj<0.05 and |lfc|>2)
```{r}
#############
#this is just a repeat of some code in the main script, but just assuring that everything necessary is here
N_DEGs05_new=row.names(res_N_Flag_new[res_N_Flag_new$padj<0.05 & !is.na(res_N_Flag_new$padj),])
length(N_DEGs05_new)
# 1353
#with abs(lfc) of 1.5
N_DEGs05_lfc1.5_new=row.names(res_N_Flag_new[res_N_Flag_new$padj<0.05 & abs(res_N_Flag_new$log2FoldChange)>1.5 & !is.na(res_N_Flag_new$padj),])
length(N_DEGs05_lfc1.5_new)
# 312
#with abs(lfc) of 2
N_DEGs05_lfc2_new=row.names(res_N_Flag_new[res_N_Flag_new$padj<0.05 & abs(res_N_Flag_new$log2FoldChange)>2 & !is.na(res_N_Flag_new$padj),])
length(N_DEGs05_lfc2_new)
#142

NLS_DEGs05_new=row.names(res_NLS_Flag_new[res_NLS_Flag_new$padj<0.05 & !is.na(res_NLS_Flag_new$padj),])
length(NLS_DEGs05_new)
# 1701
#with abs(lfc) of 1.5
NLS_DEGs05_lfc1.5_new=row.names(res_NLS_Flag_new[res_NLS_Flag_new$padj<0.05 & abs(res_NLS_Flag_new$log2FoldChange)>1.5 & !is.na(res_NLS_Flag_new$padj),])
length(NLS_DEGs05_lfc1.5_new)
# 340
#with abs(lfc) of 2
NLS_DEGs05_lfc2_new=row.names(res_NLS_Flag_new[res_NLS_Flag_new$padj<0.05 & abs(res_NLS_Flag_new$log2FoldChange)>2 & !is.na(res_NLS_Flag_new$padj),])
length(NLS_DEGs05_lfc2_new)
#175
```

Get shared DEGs from different cutoffs
```{r}
##########
#no lfc filter
intersect05_new <- intersect(N_DEGs05_new, NLS_DEGs05_new)
shared_N_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% intersect05_new, ] %>%
  dplyr::select(log2FoldChange)
N_name <- c("N_lfc")
colnames(shared_N_new) <- N_name

shared_NLS_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% intersect05_new, ] %>%
  dplyr::select(log2FoldChange)
NLS_name <- c("NLS_lfc")
colnames(shared_NLS_new) <- NLS_name
shared_both_lfc_new <- cbind(shared_N_new, shared_NLS_new)
head(shared_both_lfc_new)

#1.5
intersect05_1.5_new <- intersect(N_DEGs05_lfc1.5_new, NLS_DEGs05_lfc1.5_new)
shared_N_1.5_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% intersect05_1.5_new, ] %>%
  dplyr::select(log2FoldChange)
N_name <- c("N_lfc")
colnames(shared_N_1.5_new) <- N_name

shared_NLS_1.5_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% intersect05_1.5_new, ] %>% 
  dplyr::select(log2FoldChange)
NLS_name <- c("NLS_lfc")
colnames(shared_NLS_1.5_new) <- NLS_name
shared_both_lfc_1.5_new <- cbind(shared_N_1.5_new, shared_NLS_1.5_new)
head(shared_both_lfc_1.5_new)

#2
intersect05_2_new <- intersect(N_DEGs05_lfc2_new, NLS_DEGs05_lfc2_new)
shared_N_2_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% intersect05_2_new, ] %>%
  dplyr::select(log2FoldChange)
N_name <- c("N_lfc")
colnames(shared_N_2_new) <- N_name

shared_NLS_2_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% intersect05_2_new, ] %>%
  dplyr::select(log2FoldChange)
NLS_name <- c("NLS_lfc")
colnames(shared_NLS_2_new) <- NLS_name
shared_both_lfc_2_new <- cbind(shared_N_2_new, shared_NLS_2_new)
head(shared_both_lfc_2_new)
```

get unique degs from each part
```{r}
#########
#no lfc filter
N_only_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% N_DEGs05_new, ]
N_only_new <- subset(N_only_new, !rownames(N_only_new) %in% intersect05_new) %>%
  dplyr::select(log2FoldChange)
N_name <- c("N_lfc")
colnames(N_only_new) <- N_name
head(N_only_new)

NLS_only_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% NLS_DEGs05_new, ]
NLS_only_new <- subset(NLS_only_new, !rownames(NLS_only_new) %in% intersect05_new) %>%
  dplyr::select(log2FoldChange)
NLS_name <- c("NLS_lfc")
colnames(NLS_only_new) <- NLS_name
head(NLS_only_new)

#lfc > 1.5
N_only_1.5_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% N_DEGs05_lfc1.5_new, ]
N_only_1.5_new <- subset(N_only_1.5_new, !rownames(N_only_1.5_new) %in% intersect05_1.5_new) %>%
  dplyr::select(log2FoldChange)
colnames(N_only_1.5_new) <- N_name
head(N_only_1.5_new)

NLS_only_1.5_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% NLS_DEGs05_lfc1.5_new, ]
NLS_only_1.5_new <- subset(NLS_only_1.5_new, !rownames(NLS_only_1.5_new) %in% intersect05_1.5_new) %>%
  dplyr::select(log2FoldChange)
colnames(NLS_only_1.5_new) <- NLS_name
head(NLS_only_1.5_new)

#lfc > 2
N_only_2_new <- as.data.frame(res_N_Flag_new)[rownames(res_N_Flag_new) %in% N_DEGs05_lfc2_new, ]
N_only_2_new <- subset(N_only_2_new, !rownames(N_only_2_new) %in% intersect05_2_new) %>%
  dplyr::select(log2FoldChange)
colnames(N_only_2_new) <- N_name
head(N_only_2_new)

NLS_only_2_new <- as.data.frame(res_NLS_Flag_new)[rownames(res_NLS_Flag_new) %in% NLS_DEGs05_lfc2_new, ]
NLS_only_2_new <- subset(NLS_only_2_new, !rownames(NLS_only_2_new) %in% intersect05_2_new) %>%
  dplyr::select(log2FoldChange)
colnames(NLS_only_2_new) <- NLS_name
head(NLS_only_2_new)
```


```{r}
############
res_N_Flag_ID_new = res_N_Flag_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
head(res_N_Flag_ID_new)
nrow(res_N_Flag_ID_new)
#40420

res_NLS_Flag_ID_new = res_NLS_Flag_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
head(res_NLS_Flag_ID_new)
nrow(res_NLS_Flag_ID_new)
#40420
```

make your files into GSEA-ready files
```{r}
#all genes
###########
head(res_N_Flag_ID_new)
res_N_Flag_gsea_new <- res_N_Flag_ID_new %>%
  dplyr::select(REFSEQ, lfc_NFlag)
head(res_N_Flag_gsea_new)
res_N_Flag_gsea_new <- res_N_Flag_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(res_N_Flag_gsea_new)

head(res_NLS_Flag_ID_new)
res_NLS_Flag_gsea_new <- res_NLS_Flag_ID_new %>%
  dplyr::select(REFSEQ, lfc_NLSFlag)
head(res_NLS_Flag_gsea_new)
res_NLS_Flag_gsea_new <- res_NLS_Flag_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(res_NLS_Flag_gsea_new)
```

Shared genes
```{r}
#for the shared DEGs, I think we have to make a file for the N and NLS, even tho they're the same genes, they have different values

########
#no lfc
head(shared_N_new)
head(shared_NLS_new)

shared_N_gsea_new <- shared_N_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(shared_N_gsea_new)=paste(c("REFSEQ", "lfc_NFlag"))
shared_N_gsea_new <- shared_N_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(shared_N_gsea_new)

shared_NLS_gsea_new <- shared_NLS_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(shared_NLS_gsea_new)=paste(c("REFSEQ", "lfc_NLSFlag"))
shared_NLS_gsea_new <- shared_NLS_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(shared_NLS_gsea_new)

#lfc of >1.5
head(shared_N_1.5_new)
head(shared_NLS_1.5_new)

shared_N_gsea_1.5_new <- shared_N_1.5_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(shared_N_gsea_1.5_new)=paste(c("REFSEQ","lfc_NFlag"))
shared_N_gsea_1.5_new <- shared_N_gsea_1.5_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(shared_N_gsea_1.5_new)

shared_NLS_gsea_1.5_new <- shared_NLS_1.5_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(shared_NLS_gsea_1.5_new)=paste(c("REFSEQ","lfc_NLSFlag"))
shared_NLS_gsea_1.5_new <- shared_NLS_gsea_1.5_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(shared_NLS_gsea_1.5_new)
```

Unique genes
```{r}
#######
head(N_only_new)
head(NLS_only_new)

N_only_gsea_new <- N_only_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(N_only_gsea_new)=paste(c("REFSEQ","lfc_NFlag"))
N_only_gsea_new <- N_only_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)
head(N_only_gsea_new)

NLS_only_gsea_new <- NLS_only_new %>%
  tibble::rownames_to_column(var = "REFSEQ")
colnames(NLS_only_gsea_new)=paste(c("REFSEQ","lfc_NLSFlag"))
NLS_only_gsea_new <- NLS_only_gsea_new %>%
  left_join(cds_seq2seq) %>%
  dplyr::select(-REFSEQ)

```

Run GSEA

https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/
```{r}
#load packages
BiocManager::install("clusterProfiler")
BiocManager::install("pathview")
BiocManager::install("enrichplot")

library(clusterProfiler)

# SET THE DESIRED ORGANISM HERE
organism = "org.Hs.eg.db"
BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)

keytypes(org.Hs.eg.db)

# we want the log2 fold change 
original_gene_list_N_new <- res_N_Flag_gsea_new$lfc_NFlag
original_gene_list_NLS_new <- res_NLS_Flag_gsea_new$lfc_NLSFlag

##for N to flag
# name the vector
names(original_gene_list_N_new) <- res_N_Flag_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_N_new<-na.omit(original_gene_list_N_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_N_new = sort(gene_list_N_new, decreasing = TRUE)

#use bonferonni corrected
gse_N_new <- gseGO(geneList=gene_list_N_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)

#791 gene sets

##for NLS to flag
# name the vector
names(original_gene_list_NLS_new) <- res_NLS_Flag_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_NLS_new<-na.omit(original_gene_list_NLS_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_NLS_new = sort(gene_list_NLS_new, decreasing = TRUE)

gse_NLS_new <- gseGO(geneList=gene_list_NLS_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "BH")
#919 gene sets
```

```{r}
intersect_description_new <- intersect(gse_N_new@result$Description, gse_NLS_new@result$Description) #429 shared
```

GSEA for shared
```{r}
# we want the log2 fold change 
original_gene_list_N_shared_new <- shared_N_gsea_new$lfc_NFlag
original_gene_list_NLS_shared_new <- shared_NLS_gsea_new$lfc_NLSFlag

##for N to flag
# name the vector
names(original_gene_list_N_shared_new) <- shared_N_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_N_shared_new<-na.omit(original_gene_list_N_shared_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_N_shared_new = sort(gene_list_N_shared_new, decreasing = TRUE)

#use bonferonni corrected
gse_N_shared_new <- gseGO(geneList=gene_list_N_shared_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
            # nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)

#297 gene sets

##for NLS to flag
# name the vector
names(original_gene_list_NLS_shared_new) <- shared_NLS_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_NLS_shared_new<-na.omit(original_gene_list_NLS_shared_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_NLS_shared_new = sort(gene_list_NLS_shared_new, decreasing = TRUE)

#use bonferonni corrected
gse_NLS_shared_new <- gseGO(geneList=gene_list_NLS_shared_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)
#291

##lfc filter of 1.5
# we want the log2 fold change 
original_gene_list_N_shared_1.5_new <- shared_N_gsea_1.5_new$lfc_NFlag
original_gene_list_NLS_shared_1.5_new <- shared_NLS_gsea_1.5_new$lfc_NLSFlag

##for N to flag
# name the vector
names(original_gene_list_N_shared_1.5_new) <- shared_N_gsea_1.5_new$REFSEQ_new

# omit any NA values 
gene_list_N_shared_1.5_new<-na.omit(original_gene_list_N_shared_1.5_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_N_shared_1.5_new = sort(gene_list_N_shared_1.5_new, decreasing = TRUE)

#use bonferonni corrected
gse_N_shared_1.5_new <- gseGO(geneList=gene_list_N_shared_1.5_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
            # nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)

# no enriched gene sets

##for NLS to flag
# name the vector
names(original_gene_list_NLS_shared_1.5_new) <- shared_NLS_gsea_1.5_new$REFSEQ_new

# omit any NA values 
gene_list_NLS_shared_1.5_new<-na.omit(original_gene_list_NLS_shared_1.5_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_NLS_shared_1.5_new = sort(gene_list_NLS_shared_1.5_new, decreasing = TRUE)

#use bonferonni corrected
gse_NLS_shared_1.5_new <- gseGO(geneList=gene_list_NLS_shared_1.5_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)
#119 Gene sets
```

GSEA for unique (for the sets without lfc filter)
```{r}
# we want the log2 fold change 
original_gene_list_N_only_new <- N_only_gsea_new$lfc_NFlag
original_gene_list_NLS_only_new <- NLS_only_gsea_new$lfc_NLSFlag

##for N to flag
# name the vector
names(original_gene_list_N_only_new) <- N_only_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_N_only_new<-na.omit(original_gene_list_N_only_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_N_only_new = sort(gene_list_N_only_new, decreasing = TRUE)

#use bonferonni corrected
gse_N_only_new <- gseGO(geneList=gene_list_N_only_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)

#none

##for NLS to flag
# name the vector
names(original_gene_list_NLS_only_new) <- NLS_only_gsea_new$REFSEQ_new

# omit any NA values 
gene_list_NLS_only_new<-na.omit(original_gene_list_NLS_only_new)

# sort the list in decreasing order (required for clusterProfiler)
gene_list_NLS_only_new = sort(gene_list_NLS_only_new, decreasing = TRUE)

#use bonferonni corrected
gse_NLS_only_new <- gseGO(geneList=gene_list_NLS_only_new, 
             ont ="ALL", 
             keyType = "REFSEQ", 
             #nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             pAdjustMethod = "BH",
             verbose = TRUE, 
             OrgDb = organism)
#15 gseas
```

Now preparing the gene lists (of shared and unique DEGs) to go into Enrichr

This code is also in the refseq_conversion.R script
get the list of the shared genes (no lfc cutoff) in REFSEQ_new format
```{r}
#because we only need the gene names and not the values, we'll just use the shared list from N
head(shared_N_gsea_new)
nrow(shared_N_gsea_new)
#1192
#left_join it to have the numbers after the decimal point (because we have our genes with decimal points and those without that we used for GSEA)
sN_new1 <- shared_N_gsea_new %>%
  left_join(cds_seq2seq)
head(sN_new1)
#left_join it to have the "ISO" names
sN_new1 <- sN_new1 %>%
  left_join(cds_seq2iso_1)
head(sN_new1)
nrow(sN_new1)
#1249
#save that as a file
write.csv(sN_new1, file="./Data_Files/sN_new1.csv", quote=F)
length(unique(sN_new1$REFSEQ_new))
#1192
length(unique(sN_new1$REFSEQ))
#1249
length(unique(sN_new1$ISO))
#1249
```

Now do the same with the unique genes (no lfc cutoff)
```{r}
#N-only genes (no lfc cutoff)
head(N_only_gsea_new)
nrow(N_only_gsea_new)
#161
oN_new1 <- N_only_gsea_new %>%
  left_join(cds_seq2seq)
head(oN_new1)
oN_new1 <- oN_new1 %>%
  left_join(cds_seq2iso_1)
head(oN_new1)
nrow(oN_new1)
#169
#save that as a file
write.csv(oN_new1, file="./Data_Files/oN_new1.csv", quote=F)
length(unique(oN_new1$REFSEQ_new))
#161
length(unique(oN_new1$REFSEQ))
#169
length(unique(oN_new1$ISO))
#169

#NLS-only genes (no lfc cutoff)
head(NLS_only_gsea_new)
nrow(NLS_only_gsea_new)
#509
oNLS_new1 <- NLS_only_gsea_new %>%
  left_join(cds_seq2seq)
head(oNLS_new1)
oNLS_new1 <- oNLS_new1 %>%
  left_join(cds_seq2iso_1)
head(oNLS_new1)
nrow(oNLS_new1)
#524
#save that as a file
write.csv(oNLS_new1, file="./Data_Files/oNLS_new1.csv", quote=F)
length(unique(oNLS_new1$REFSEQ_new))
#509
length(unique(oNLS_new1$REFSEQ))
#524
length(unique(oNLS_new1$ISO))
#524
```

Do the same for the shared DEGS with 1.5 lfc cutoff
```{r}
head(shared_N_gsea_1.5_new)
nrow(shared_N_gsea_1.5_new)
#217
sN_1.5_new1 <- shared_N_gsea_1.5_new %>%
  left_join(cds_seq2seq)
head(sN_1.5_new1)
#left_join it to have the "ISO" names
sN_1.5_new1 <- sN_1.5_new1 %>%
  left_join(cds_seq2iso_1)
head(sN_1.5_new1)
nrow(sN_1.5_new1)
#226
#save that as a file
write.csv(sN_1.5_new1, file="./Data_Files/sN_1.5_new1.csv", quote=F)
length(unique(sN_1.5_new1$REFSEQ_new))
#217
length(unique(sN_1.5_new1$REFSEQ))
#226
length(unique(sN_1.5_new1$ISO))
#226
```

In order to run these different groups of DEGs through enrichr, the REFSEQ gene names have to be converted to Entrez gene symbols

I use this portal: https://www.biotools.fr/human/refseq_symbol_converter
The REFSEQ variety that works best is the one we have labeled "ISO" here 
The best way to load all the ISOs you want into the portal for Entrez conversion is to open the .csv files that you just generated in excel, then copy the entire column of ISO into the portal

You can then copy the list of Entrez symbols from the portal into Enrichr:
https://maayanlab.cloud/Enrichr/

Because we are interested in the Enrichr outputs of the shared DEGs (no lfc cutoff), I added the Entrez symbols of these DEGs to a column of the csvs as well to make "sN_new1_symbol.csv", which I uploaded into Data_Files

We are interested in the results from the "TF Perturbations Followed by Expression" group output from Enrichr

Going to read that new file in
```{r}
#make a variable for the genes from the shared terms that overlap with the Enrichr rela kd geneset
relakd <- c("SRP19", "POP7", "RTCB", "ACTB", "TOMM22", "PSMD8", "CDC20", "CHCHD2", "PSMD7", "GABARAPL2", "SUPT16H", "ATP1B3", "COX6B1", "RRAGA", "MTHFD1", "UQCRC1", "SRSF2", "SUCLG1", "UQCRC2", "SNRPF", "SELENOT", "SNRPB", "ANP32A", "PSMD14", "DDX1", "TXN", "PRDX3", "PRDX2", "PRDX4", "PCBP1", "CYC1", "HSPA9", "XRCC5", "DHCR24", "TBCA", "PRDX6", "LSM3", "AIMP2", "CNIH1", "UQCRQ", "BAMBI", "CYCS", "RNPS1", "COX7B", "LAMC1", "HMGB1", "YBX1", "RBM3", "MRPL3", "YWHAQ", "C1QBP", "DHX15", "UQCRFS1", "ELOB", "RAB8A", "COX8A", "PARP1", "HSBP1", "ZFR", "TUBG1", "MRPS7", "DDOST", "APRT", "PSMA2", "DAD1", "NDUFS6", "CKS2", "PABPC1", "SLC25A5", "ARF5", "VAMP3", "NDUFB8", "SF3B3", "ECHS1", "NDUFB6", "UQCR10", "COX5A", "SRM", "PSMB7", "ATP5F1B", "PSMB3", "POLR2C", "PSMB1", "EMC8", "NDUFA9", "TIMM8B", "GINS2", "SF3A1", "MARCKSL1", "TUBB4B", "HSPE1", "EIF3K", "SCD", "PSMC2", "DRG1", "ATP5PO", "RBMX")
#read in csv where you've added the Entrez symbols
sN_new2 <- read.csv(file="./Data_Files/sN_new1_symbol.csv", row.names = 1)
head(sN_new2)
#extract the relakd terms
sN_rel <- sN_new2 %>%
  subset(SYMBOL %in% relakd)
nrow(sN_rel)
#101

#now get the lfc values for the shared NLS (because the file we made earlier just had the N)
head(shared_NLS_gsea_new)
nrow(shared_NLS_gsea_new)
#1192
sNLS_new1 <- shared_NLS_gsea_new %>%
  left_join(cds_seq2seq)
head(sNLS_new1)
sNLS_new1 <- sNLS_new1 %>%
  left_join(cds_seq2iso_1)
head(sNLS_new1)
nrow(sNLS_new1)
#1249
#save that as a file
write.csv(sNLS_new1, file="./Data_Files/sNLS_new1.csv", quote=F)
length(unique(sNLS_new1$REFSEQ_new))
#1192
length(unique(sNLS_new1$REFSEQ))
#1249
length(unique(sNLS_new1$ISO))
#1249

#now add the sNLS lfc values sN_rel
sN_NLS_rel <- sN_rel %>%
  left_join(sNLS_new1)
nrow(sN_NLS_rel)
#101
```

Look at the expression of these genes of interest in the N samples and the NLS samples

In order to make a heatmap, I will want the expression value across all the samples, not just the average lfc per treatment, and I might as well include the flag samples as well
```{r}
rld_new_named <- read.csv(file="./Data_Files/mingers_rldall_new_named_24.csv", row.names = 1)

#make each symbol unique
sN_NLS_rel_un <- sN_NLS_rel %>%
  mutate(SYMBOL = make.names(SYMBOL, unique = TRUE))

sN_NLS_rel_rld <- sN_NLS_rel_un %>%
  left_join(rld_new_named) %>%
  select(-lfc_NFlag, -lfc_NLSFlag, -REFSEQ_new, -REFSEQ, -ISO) %>%
  column_to_rownames(var = "SYMBOL")
head(sN_NLS_rel_rld)

#make a heatmap
sN_rel_means = apply(sN_NLS_rel_rld, 1, mean)
explc_sN_rel = sN_NLS_rel_rld-sN_rel_means

sN_rel_hm <- pheatmap(as.matrix(explc_sN_rel),
         color=rev(colorRampPalette(brewer.pal(n=11, name="RdBu"))(20)),
         cluster_rows = T, show_rownames = T, fontsize_row=8, fontsize_col=10, treeheight_row = 10, treeheight_col = 10,
         cluster_cols = T, show_colnames = T, angle_col= "315", cellheight = 8, cellwidth = 14,
         breaks=c(seq(from=-1,to=1,by=0.1)),
         scale="none",
         legend=T)
#saved as sN_rel_hm_060624 as 12x5 portrait
sN_rel_hm_nonames <- pheatmap(as.matrix(explc_sN_rel),
         color=rev(colorRampPalette(brewer.pal(n=11, name="RdBu"))(20)),
         cluster_rows = T, show_rownames = F,
         cluster_cols = T, show_colnames = T, angle_col= "315",
         breaks=c(seq(from=-1,to=1,by=0.1)),
         scale="none",
         legend=T)
#saved as sN_rel_hm_nonames as 5x4 portrait
```

Making bargraph of the terms of interest downloaded from enrichr
```{r}
##load in the entire TF_Perturbations_Followed_by_Expression dataset
TF_pert <- read.delim("./Data_Files/TF_Perturbations_Followed_by_Expression_table.txt")
TF_pert_top10 <- TF_pert[1:10,]
##the whole TF_pert_top10 could be a sup table

TF_pert_top10 <- TF_pert_top10 %>%
  mutate(Term = gsub("RELA KD HUMAN GSE1676 CREEDSID GENE 1787 UP","RELA KD GSE1676", Term)) %>%
  mutate(Term = gsub("HNF1B OE HUMAN GSE3308 CREEDSID GENE 1750 UP","HNF1B OE GSE3308", Term)) %>%
  mutate(Term = gsub("TP53 KD HUMAN GSE1676 CREEDSID GENE 1784 UP","TP53 KD GSE1676", Term)) %>%
  mutate(Term = gsub("HNF1A OE HUMAN GSE3308 CREEDSID GENE 279 DOWN","HNF1A OE GSE3308", Term)) %>%
  mutate(Term = gsub("DOT1L INHIBITION HUMAN GSE29828 CREEDSID GENE 1239 DOWN","DOT1L INHIBITION GSE29828", Term)) %>%
  mutate(Term = gsub("NFKB1 INACTIVATION HUMAN GSE20667 CREEDSID GENE 2577 DOWN","NFKB1 INACTIVATION GSE20667.2577", Term)) %>%
  mutate(Term = gsub("SOX4 KD HUMAN GSE4225 CREEDSID GENE 64 DOWN","SOX4 KD GSE4225", Term)) %>%
  mutate(Term = gsub("MYB OE HUMAN GSE2815 CREEDSID GENE 1740 UP","MYB OE GSE2815", Term)) %>%
  mutate(Term = gsub("NFKB1 INACTIVATION HUMAN GSE20667 CREEDSID GENE 2576 DOWN","NFKB1 INACTIVATION GSE20667.2576", Term)) %>%
  mutate(Term = gsub("MYC KNOCKIN MOUSE GSE39443 CREEDSID GENE 1847 UP","MYC KNOCKIN GSE39443", Term))

TF_pert_top10$Term <- factor(TF_pert_top10$Term, levels = c("RELA KD GSE1676", "HNF1B OE GSE3308", "TP53 KD GSE1676", "HNF1A OE GSE3308", "DOT1L INHIBITION GSE29828", "NFKB1 INACTIVATION GSE20667.2577", "SOX4 KD GSE4225", "MYB OE GSE2815", "NFKB1 INACTIVATION GSE20667.2576", "MYC KNOCKIN GSE39443"))

TF_pert_top10$Term <- factor(TF_pert_top10$Term, levels = c("MYC KNOCKIN GSE39443", "NFKB1 INACTIVATION GSE20667.2576", "MYB OE GSE2815", "SOX4 KD GSE4225", "NFKB1 INACTIVATION GSE20667.2577", "DOT1L INHIBITION GSE29828", "HNF1A OE GSE3308", "TP53 KD GSE1676", "HNF1B OE GSE3308", "RELA KD GSE1676"))

#make a bargraph
tf_pert_bar <- ggplot(data=TF_pert_top10, aes(x=Term, y=Combined.Score))+
  geom_bar(position="stack", stat="identity", color="black", alpha=0.6)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        panel.background = element_rect(color="black", fill="white"),
        axis.text = element_text(colour="black", size=10), 
        axis.title.x = ggtext::element_markdown(face="bold", size=12),
        axis.title.y = ggtext::element_markdown(face="bold",size=12),
        panel.grid.major = element_line(colour = "gray80", linewidth = 0.5, linetype="dotted")) 
#saved as tf_pert_bar as landscape 4x4

tf_pert_bar_h <- ggplot(data=TF_pert_top10, aes(x=Combined.Score, y=Term))+
  geom_bar(position="stack", stat="identity", color="black", alpha=0.6)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        panel.background = element_rect(color="black", fill="white"),
        axis.text = element_text(colour="black", size=10), 
        axis.title.x = ggtext::element_markdown(face="bold", size=12),
        axis.title.y = ggtext::element_markdown(face="bold",size=12),
        panel.grid.major = element_line(colour = "gray80", linewidth = 0.5, linetype="dotted")) 
#saved as tf_pert_bar_horizontal as landscape 4x3.5
```



